pkg/kubectl/resource_printer_test.go

```diff
@@ -276,12 +284,17 @@ func TestUnknownTypePrinting(t *testing.T) {
 
 func TestTemplateEmitsVersionedObjects(t *testing.T) {
 	// kind is always blank in memory and set on the wire
-	printer, err := NewTemplatePrinter([]byte(`{{.kind}}`), testapi.Version(), api.Scheme)
+	printer, err := NewTemplatePrinter([]byte(`{{.kind}}`))
 	if err != nil {
 		t.Fatalf("tmpl fail: %v", err)
 	}
+	obj, err := api.Scheme.ConvertToVersion(&api.Pod{}, "v1beta1")
```

Comment:
The underlying problem was that a printer had to deal with objects of different api versions, so no single printer that only accepted a single printer would work.